### ISSUE TITLE
Things that need doing

### ISSUE THREAD

**Issue by colbiedison** [colbiedison/compactmachines-fabric#3]
**Tunnels**
The main things needing work are the Tunnels. see [feature/tunnels](https://github.com/colbiedison/compactmachines-fabric/tree/feature/tunnels)
Item tunnels are mostly working, but could probably be done much better.
Other types (energy, fluid, etc) are needed.

**Update to 1.19**

**Comment by TheDrawingCoder-Gamer** [comment]
What mods do you test tunnels with? I fixed the dimension on 1.18.2 (the version I use for testing, it mostly works afaik), but I can't place tunnels.

**Comment by TheDrawingCoder-Gamer** [comment]
... and manually placing the tunnel block crashes the game + cooks the world

**Comment by colbiedison** [comment]
I've only tested the tunnels with vanilla hoppers/chests. I don't remember if the tunnel items in the creative menu work correctly. I think I mainly spawned them with `/give <player> compactmachines_fabirc:tunnel{type:"Item"}` or something like that.

**Comment by colbiedison** [comment]
https://cdn.discordapp.com/attachments/383488878935408650/968996633171853412/simplescreenrecorder-2022-04-27_17.05.49.mp4
https://cdn.discordapp.com/attachments/383488878935408650/969038341628637204/2022-04-27_19-48-27.mp4

**Comment by TheDrawingCoder-Gamer** [comment]
yes, i also had to manually spawn them. I edited the item group so it omits the tunnel wall block as it cooked my world, and omits the unknown tunnel and replaces it with the 3 variants

**Comment by colbiedison** [comment]
The machine wall blocks (unbreakable ones and tunnel ones) have NBT data identifying which machine they belong to. If you need to place a wall block inside a machine and have it work, you need to either manually set its `parent`/`id` field (i don't remember which), or ctrl + pick block to copy its data.

**Comment by TheDrawingCoder-Gamer** [comment]
https://github.com/TheDrawingCoder-Gamer/compactmachines-fabric/projects?type=beta here is the current stuff I have... redstone wires don't block update upon going through a tunnel. it was a nightmare to figure out why they wouldn't go through in the first place.

**Comment by TheDrawingCoder-Gamer** [comment]
Ok, it all supports the transfer api now. Only issue is connections don't work properly sometimes. I.e. if both sides are a pipe, they won't connect to each other. I got rid of the energy tunnel. Based off of compact machines 3, item, fluid, and energy all go through the item tunnel
